src/main/java/eu/europeana/metis/sandbox/service/metrics/DatasetMetrics.java

```diff
@@ -0,0 +1,23 @@
+package eu.europeana.metis.sandbox.service.metrics;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import java.util.Map;
+
+@JsonInclude(Include.NON_EMPTY)
+public class DatasetMetrics {
```

Comment:
Why this custom object? Can't we use the MeterRegistry from Micrometer or not?  
Also if we do, I think then the apm agent could potentially pick them up but not sure how that works at this point.  
Check here https://www.elastic.co/guide/en/apm/agent/java/current/metrics.html

Reply:
Here is used the custom object suggested by Springboot actuator when adding a custom endpoint. So when adding this property the metrics can be exported to elastic.
```
management:
  metrics:
    export:
      elastic:
        host: "https://elastic.example.com:8086"
```

Reply:
First of all, we don't send anything directly to elasticsearch, it first goes to a proxy, logstash or apm(currently the same machine but different ports).
Okay, but we'll need to make it work as a time series database which I think it might not work with the above(maybe it will just index the documents or so), not sure though.
What I meant is that if we use the objects that the APM understands, then the apm agent we have connected already might pick those information automatically.
We'll have to test those assumptions though, since what I'm saying is just how I understood it from the docs as I haven't performed that in the past.